website/docs/proposals/20250630-kaito-cli.md

```diff
@@ -0,0 +1,526 @@
+
+| **Use Case**                             | **Example**                                                                                                      |
+| ---------------------------------------- | ---------------------------------------------------------------------------------------------------------------- |
+| Platform team managing multiple clusters | `kaitoctl status --all --cluster us-east,us-west`                                                                |
+| CI/CD offline manifest render            | `kaitoctl deploy --n workspace-phi --model phi --dry-run > ws.yaml`                                              |
+| Ops analyst estimating GPU cost          | `kaitoctl cost --workspace gen-answers`                                                                          |
+| LoRA tuning with progress bar            | `kaitoctl tune --name social-sentiment --interactive`                                                            |
+| Distributed inference model deployment   | `kaitoctl deploy --name workspace-mixtral --model mixtral --distributed.enabled=true --distributed.partitions=4` |
+
+**Pros:**
+
+- Full UX control: CLI prompts, spinners, YAML preview, etc.
+- Works without kubectl — just kubeconfig
+- Can support offline use cases and CI manifest staging
+- Supports **multi-cluster** targeting from a single command
+
+**Cons:**
+
+- Slightly larger binary with more dependencies
+- Must manage auth lifecycle (client-go, OIDC, token refresh)
+- Requires separate install path (brew, apt, etc.)
+
+### Decision Drivers
+
+1. **Adoption barrier**: If speed of rollout is top priority, Option A excels
+2. **Feature depth**: For advanced, multi-cluster, or offline workflows, Option B is preferable
+3. **Resourcing**: Building both offers the richest UX but doubles initial engineering effort
+
+## Core Features and Commands
+
+### Workspace Management
+
+#### Deployment Commands
+
+```bash
+# Basic inference deployment
+kubectl kaito deploy --name my-workspace --model phi-3.5-mini-instruct --instance-type Standard_NC6s_v3
+
+# Deploy with specific runtime
+kubectl kaito deploy --name my-workspace --model phi-3-mini --runtime vllm
+kubectl kaito deploy --name my-workspace --model falcon-7b --runtime transformers
+
+# Deploy with adapters
+kubectl kaito deploy --name my-workspace --model phi-3-mini \
+  --adapter my-adapter:v1 --adapter-strength 0.8
+
+# Deploy distributed inference
+kubectl kaito deploy --name my-distributed --model llama-3.3-70b \
+  --distributed --nodes 4 --runtime vllm
+
+# Deploy with feature gates
+kubectl kaito deploy --name my-workspace --model phi-3-mini --feature-gates vLLM=true,ensureNodeClass=false
+```
+
+#### Management Commands
+
+```bash
+# Scale inference replicas
+kubectl kaito scale workspace/my-workspace --replicas 3
+
+# Status and monitoring
+kubectl kaito status workspace/my-workspace
```

Comment:
If native k8s support this, should we add it to this tool?

I think this way also works.
```
kubectl scale --replicas=3 workspace/my-workspace
kubectl describe workspace/my-workspace
```